tests/direct-proxy_test.go

```diff
@@ -30,114 +30,30 @@ import (
-		httpClient = &http.Client{
-			Transport: transport,
-		}
-
-		serviceAccount = &core.ServiceAccount{
-			ObjectMeta: metav1.ObjectMeta{
-				GenerateName: "test-service-account-",
-				Namespace:    testNamespace,
-			},
-		}
-		serviceAccount, err := ApiClient.CoreV1().ServiceAccounts(testNamespace).Create(context.TODO(), serviceAccount, metav1.CreateOptions{})
-		Expect(err).ToNot(HaveOccurred())
-		DeferCleanup(func() {
-			err := ApiClient.CoreV1().ServiceAccounts(testNamespace).Delete(context.TODO(), serviceAccount.Name, metav1.DeleteOptions{})
-			if err != nil && !errors.IsNotFound(err) {
-				Expect(err).ToNot(HaveOccurred())
-			}
-			serviceAccount = nil
+		UpdateConfigMap(func(configMap *core.ConfigMap) {
+			configMap.Data[config.SaTokensFilename] = "false"
 		})
-
-		role := &rbac.Role{
-			ObjectMeta: metav1.ObjectMeta{
-				GenerateName: "test-role-",
-				Namespace:    testNamespace,
-			},
-			Rules: []rbac.PolicyRule{{
-				APIGroups: []string{kubevirtcorev1.SubresourceGroupName},
-				Resources: []string{"virtualmachineinstances/vnc"},
-				Verbs:     []string{"get"},
-			}},
-		}
-		role, err = ApiClient.RbacV1().Roles(testNamespace).Create(context.TODO(), role, metav1.CreateOptions{})
-		Expect(err).ToNot(HaveOccurred())
 		DeferCleanup(func() {
-			err := ApiClient.RbacV1().Roles(testNamespace).Delete(context.TODO(), role.Name, metav1.DeleteOptions{})
-			if err != nil && !errors.IsNotFound(err) {
-				Expect(err).ToNot(HaveOccurred())
-			}
+			RevertToOriginalConfigMap()
 		})
 
-		roleBinding = &rbac.RoleBinding{
-			ObjectMeta: metav1.ObjectMeta{
-				GenerateName: "test-clusterrolebinding-",
-				Namespace:    testNamespace,
-			},
-			Subjects: []rbac.Subject{{
-				Kind:      "ServiceAccount",
-				Name:      serviceAccount.Name,
-				Namespace: serviceAccount.Namespace,
-			}},
-			RoleRef: rbac.RoleRef{
-				APIGroup: rbac.GroupName,
-				Kind:     "Role",
-				Name:     role.Name,
-			},
-		}
-		roleBinding, err = ApiClient.RbacV1().RoleBindings(testNamespace).Create(context.TODO(), roleBinding, metav1.CreateOptions{})
-		Expect(err).ToNot(HaveOccurred())
-		DeferCleanup(func() {
-			err := ApiClient.RbacV1().RoleBindings(testNamespace).Delete(context.TODO(), roleBinding.Name, metav1.DeleteOptions{})
-			if err != nil && !errors.IsNotFound(err) {
-				Expect(err).ToNot(HaveOccurred())
-			}
-			roleBinding = nil
-		})
+		// It can take a while before kubernetes propagates the ConfigMap change to the pod.
+		waitUntilTokenGenerationMethod(api.LocalTokenGeneration)
 
 		tokenRequest := &authnv1.TokenRequest{}
-		tokenRequest, err = ApiClient.CoreV1().ServiceAccounts(testNamespace).CreateToken(context.TODO(), serviceAccount.Name, tokenRequest, metav1.CreateOptions{})
+		tokenRequest, err := ApiClient.CoreV1().ServiceAccounts(testNamespace).CreateToken(context.TODO(), serviceAccountName, tokenRequest, metav1.CreateOptions{})
 		Expect(err).ToNot(HaveOccurred())
 
 		saToken = tokenRequest.Status.Token
```

Comment:
Personally I'd prefer to name it `serviceAccountToken`, and in all related places instead of `sa`. Or just keep it as `token`.

Reply:
That's not in scope of this PR. We can post another one.